src/Qowaiv.Validation.Xml/XmlValidationHandler.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace Qowaiv.Validation.Xml;
+
+/// <summary>An <see cref="ValidationEventHandler"/> that collects all issues as <see cref="IValidationMessage"/>'s.</summary>
+public class XmlValidationHandler
+{
+    /// <summary>The collection of validation messages.</summary>
+    public IReadOnlyCollection<IValidationMessage> Messages { get; } = new List<IValidationMessage>();
+
+    /// <summary>The <see cref="ValidationEventHandler"/> delegate.</summary>
+    public ValidationEventHandler Validate => (sender, e) => OnIssue(sender, e);
+
+    private void OnIssue(object? sender, ValidationEventArgs e)
+        => ((List<IValidationMessage>)Messages)
+        .Add(ValidationMessage.For(e.Severity.ToValidationSeverity(), e.Message, PropertyName(sender)));
+
+    [Pure]
+    private static string? PropertyName(object? sender)
+        => sender switch
+        {
+            XElement x => x.AbsoluteXPath(),
```

Comment:
Can you do `XElement x or XAttribute x => x.AbsoluteXPath()` here? Or is that not possible because of the different types.

Reply:
No that is not possible. I could define this extension method on a shared base, but than I would have to do a switch in that code. That would make things worse.